spi-device.d.ts

```diff
@@ -0,0 +1,46 @@
+export type MODE0 = 0;
+export type MODE1 = 1;
+export type MODE2 = 2;
+export type MODE3 = 3;
+
+export type SpiMode = MODE0 | MODE1 | MODE2 | MODE3;
+
+export type SpiOptions = {
+    mode?: number, // 2-bit, MODE0, MODE1, MODE2, or MODE3, default MODE0
+    chipSelectHigh?: boolean, // true for active high chip select, default false
+    lsbFirst?: boolean, // true for least significant bit first transfer, default false
+    threeWire?: boolean, // true for shared MISO/MOSI signals, default false
+    loopback?: boolean, // true for loopback mode, default false
+    noChipSelect?: boolean, // true for 1 device per bus, no chip select, default false
+    ready?: boolean, // true if device pulls low to pause, default false
+    bitsPerWord?: number, // 8-bit, device word size, default 8
+    maxSpeedHz?: number // 32-bit, device clock frequency in Hertz, default system specific
+}
+
+export type SpiMessage = {
```

Comment:
An SPI message is an array of one or more read+write transfers, not a single read+write transfer.

Reply:
Hmm okay... shall it be: `export type SpiMessage[] = {...}`?

Reply:
I don't know as I'm not familiar wit TypeScript.

The following represents a single SPI transfer:
```ts
{
    byteLength: number, // 32-bit, the number of bytes to transfer
    sendBuffer?: Buffer, // transmit data
    receiveBuffer?: Buffer, // receive data
    speedHz?: number, // 32-bit, override of the device's clock frequency in Hertz
    microSecondDelay?: number, // 16-bit, delay after the last bit transfer before optionally deselecting the device before the next transfer, default 0
    bitsPerWord?: number, // 8-bit, override of the device's wordsize
    chipSelectChange?: boolean, // true to deselect device before starting the next transfer, default false
}
```
This isn't an SPI message for spi-device. For spi-device, an SPI message is an array of SPI transfers.

Please run some tests with a hardware device to verify that the correct thing is being done here.

Reply:
Hi @fivdi ,

What about something like:

```typescript
export type SpiTransfer = {
    byteLength: number, // 32-bit, the number of bytes to transfer
    sendBuffer?: Buffer, // transmit data
    receiveBuffer?: Buffer, // receive data
    speedHz?: number, // 32-bit, override of the device's clock frequency in Hertz
    microSecondDelay?: number, // 16-bit, delay after the last bit transfer before optionally deselecting the device before the next transfer, default 0
    bitsPerWord?: number, // 8-bit, override of the device's wordsize
    chipSelectChange?: boolean, // true to deselect device before starting the next transfer, default false
}

export type SpiMessage = SpiTransfer[];

```

Sorry, I'm also not quite familiar with TypeScript. What I understood is, this kind of style works, while my last commit (https://github.com/huming2207/spi-device/blob/master/spi-device.d.ts#L20) should work as well, unless you want to explicitly use a single `SpiTransfer` object for some reasons later.

By the way, I will test it tomorrow. I will tell you the result.

Regards,
Jackson